src/backend/replication/gp_replication.c

```diff
@@ -587,7 +587,11 @@ GetMirrorStatus(FtsResponse *response, bool *ready_for_syncrep)
 	LWLockRelease(SyncRepLock);
 
 	if (!response->IsMirrorUp)
+	{
 		response->RequestRetry = is_probe_retry_needed();
+		if (!response->RequestRetry)
+			SIMPLE_FAULT_INJECTOR("replication_mirror_down");
+	}
```

Comment:
Maybe we can use `wait_for_mirror_down` (see setup.sql) instead of creating a new fault injector? To wait for the FTS to mark the mirror as down.

Reply:
Seems to be working, changed to use `wait_for_mirror_down`